src/lesson/router.ts

```diff
@@ -11,87 +11,188 @@ import * as handlers from './handlers';
+   *       500:
+   *         description: Server error during lesson creation
+   */
+  router.post("/", controller.createLesson);
 
-    return router;
+  /**
+   * @swagger
+   * /lesson:
+   *   get:
+   *     summary: Get all lessons
+   *     tags: [Lesson]
+   *     responses:
+   *       200:
+   *         description: A list of lessons
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: array
+   *               items:
+   *                 $ref: '#/components/schemas/Lesson'
+   *       500:
+   *         description: Server error
+   */
+  router.get("/", controller.getLessons);
+
+  /**
+   * @swagger
+   * /lesson/delete/{id}:
+   *   delete:
+   *     summary: Delete a lesson by its ID
+   *     tags: [Lesson]
+   *     parameters:
+   *       - in: path
+   *         name: id
+   *         required: true
+   *         schema:
+   *           type: string
+   *         description: ID of the lesson to delete
+   *     responses:
+   *       200:
+   *         description: Lesson successfully deleted
+   *         content:
+   *           application/json:
+   *             schema:
+   *               type: object
+   *               properties:
+   *                 message:
+   *                   type: string
+   *                   example: Lesson with id 123 deleted successfully.
+   *       404:
+   *         description: Lesson not found
+   *       400:
+   *         description: Invalid ID format
+   *       500:
+   *         description: Server error
+   */
+  router.delete("/delete/:id", controller.deleteLesson);
+
+  /**
+   * @swagger
+   * /lesson/update/{id}:
+   *   put:
+   *     summary: Update a lesson by its ID
+   *     tags: [Lesson]
+   *     parameters:
+   *       - in: path
+   *         name: id
+   *         required: true
+   *         schema:
+   *           type: string
+   *         description: ID of the lesson to update
+   *     requestBody:
+   *       required: true
+   *       content:
+   *         application/json:
+   *           schema:
+   *             type: object
+   *             properties:
+   *               title:
+   *                 type: string
+   *                 description: The new title of the lesson
+   *               summary:
+   *                 type: string
+   *                 description: The new summary of the lesson
+   *           example:
+   *             title: Updated Lesson Title
+   *             summary: Updated summary content.
+   *     responses:
+   *       200:
+   *         description: Lesson successfully updated
+   *         content:
+   *           application/json:
+   *             schema:
+   *               $ref: '#/components/schemas/Lesson'
+   *       404:
+   *         description: Lesson not found
+   *       400:
+   *         description: Invalid ID format
+   *       500:
+   *         description: Server error
+   */
+  router.put("/update/:id", controller.updateLesson);
```

Comment:
same as delete, no need for prefix
```suggestion
  router.put("/:id", controller.updateLesson);
```